ydb/core/kqp/provider/yql_kikimr_exec.cpp

```diff
@@ -1628,22 +1636,69 @@ class TKiSinkCallableExecutionTransformer : public TAsyncCallbackTransformer<TKi
                                     return SyncError();
                                 }
                             }
-                        } else {
-                            ctx.AddError(TIssue(ctx.GetPosition(nameNode.Pos()),
-                                TStringBuilder() << "Unknown add index setting: " << name));
+                        } else if (name == "indexSettings") {
+                            YQL_ENSURE(add_index->type_case() == Ydb::Table::TableIndex::kGlobalVectorKmeansTreeIndex);
+                            auto& protoVectorSettings = *add_index->mutable_global_vector_kmeans_tree_index()->mutable_vector_settings();
+                            auto indexSettings = columnTuple.Item(1).Cast<TCoAtomList>();
+                            YQL_ENSURE(indexSettings.Maybe<TCoNameValueTupleList>());
+                            for (const auto& vectorSetting : indexSettings.Cast<TCoNameValueTupleList>()) {
+                                YQL_ENSURE(vectorSetting.Value().Maybe<TCoAtom>());
+                                if (vectorSetting.Name().Value() == "distance") {
+                                    auto parseEnum = [] (const TString distance) {
+                                        if (distance == "cosine")
+                                            return Ydb::Table::VectorIndexSettings_Distance_DISTANCE_COSINE;
+                                        else if (distance == "manhattan")
+                                            return Ydb::Table::VectorIndexSettings_Distance_DISTANCE_MANHATTAN;
+                                        else if (distance == "euclidean")
```

Comment:
Is `distance` always in lower case? Or maybe we want some case insensitive equals check? 

This comment also for other settings

Reply:
The distance string constants are taken from comments 
https://github.com/ydb-platform/ydb/pull/6602/files#diff-d65ef26e8c431c0f100d47987db76a78d4132cdb933a7e3f9d3c5860e928f8c5R1157